test/common/router/config_impl_test.cc

```diff
@@ -1651,52 +1686,116 @@ TEST_F(RouterMatcherHashPolicyTest, HashIpv6DifferentAddresses) {
   }
 }
 
+TEST_F(RouterMatcherHashPolicyTest, QueryParameterHash) {
+  firstRouteHashPolicy()->mutable_query_parameter()->set_key("hash");
+  uint64_t hash_1;
+  uint64_t hash_2;
+  {
+    Http::TestHeaderMapImpl headers = genHeaders("www.lyft.com", "/foo?hash=bluh", "GET");
+    const auto hash_policy = config().route(headers, 0)->routeEntry()->hashPolicy();
+    hash_1 = hash_policy->generateHash(nullptr, headers, add_cookie_nop_).value();
+  }
+  {
+    Http::TestHeaderMapImpl headers = genHeaders("www.lyft.com", "/foo?hash=meh", "GET");
+    const auto hash_policy = config().route(headers, 0)->routeEntry()->hashPolicy();
+    hash_2 = hash_policy->generateHash(nullptr, headers, add_cookie_nop_).value();
+  }
+  EXPECT_NE(hash_1, hash_2);
+}
+
+TEST_F(RouterMatcherHashPolicyTest, QueryParameterHashNoSuchKey) {
+  Http::TestHeaderMapImpl headers = genHeaders("www.lyft.com", "/foo?hash=bluh&bluh=eh", "GET");
+
+  firstRouteHashPolicy()->mutable_query_parameter()->set_key("meh");
+  const auto hash_policy = config().route(headers, 0)->routeEntry()->hashPolicy();
+  EXPECT_FALSE(hash_policy->generateHash(nullptr, headers, add_cookie_nop_));
+}
+
+TEST_F(RouterMatcherHashPolicyTest, HostPathHash) {
```

Comment:
I don't know what the expected behavior is, but you should probably have a test for HostPath with different port numbers.

Reply:
Will do (if we ever add back hostpath hash method the way I expected it to be).